src/controllers/permissions.js

```diff
@@ -0,0 +1,64 @@
+import { checkIfExistsInDb } from '../utils/searchDb';
+import { resource, permission } from '../models';
+/**
+ * @class
+ * @description Class based Permissions Controller
+ */
+class Permissions {
+  /**
+   * @description Sets the permission for a given role to a particular resource
+   * @static
+   * @param {object} req object
+   * @param {object} res object
+   * @param {method} next method
+   * @returns {object } Updated Role information
+   * @memberof Permissions
+   * @type {object}
+   */
+  static async setPermissions(req, res, next) {
+    let updatedPermissions;
+    const { roleId } = req.params;
+    const { resourceId, ...permissions } = req.body;
```

Comment:
why are you spreading permissions here?

Reply:
The Super Admin can set more than one permission for a role on a resource. The input is
{
  resourceId: 3,
  add: true,
  read: true
}
So I am using the spread operators to get all possible permission input